File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        a = 0
        b = int(c ** 0.5)

        while a <= b:
            sq_sum = a * a + b * b

            if sq_sum == c:
                return True
            elif sq_sum < c:
                a += 1  # need a larger sum
            else:
                b -= 1  # need a smaller sum

        return False

File: 633-sum-of-square-numbers/sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(long long c) {
        long long a = 0;
        long long b = (long long)sqrt(c);

        while (a <= b) {
            long long sqSum = a * a + b * b;

            if (sqSum == c) {
                return true;
            } else if (sqSum < c) {
                a++; // need a larger sum
            } else {
                b--; // need a smaller sum
            }
        }

        return false;
    }
};